### PR TITLE
Properly handle arrays where index can be negative

### PR DESCRIPTION
**Pull request by Dr15Jones**
Instead of relying on remembering to always adding an offset to the
index being passed to the array, we now use a helper class which
always does the offsetting for us. The helper also properly initializes
the size of the array to be able to contain the min and max bounds.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@davidlange6 @smuzaffar why does this show up as new-package-pending when the code I changed is already in the release?

**Comment by Dr15Jones** [comment]
assign l1

**Comment by davidlt** [comment]
The package was never added to https://github.com/cms-sw/cms-bot/blob/master/categories.py

**Comment by smuzaffar** [comment]
@Dr15Jones , L1Trigger/L1TTwinMux is now added under l1.

**Comment by davidlange6** [comment]
please test

**Comment by Dr15Jones** [comment]
@smuzaffar Thanks!

**Comment by Dr15Jones** [comment]
The only difference is in the number of MessageLogger Errors and Warnings (I think they decreased) in the workflow 10224.0.

**Comment by Dr15Jones** [comment]
@davidlange6 this should fix the crashes in the aarch64 crashes in the IBs.

File: L1Trigger/L1TTwinMux/src/RPCHitCleaner.cc [hunk 1]
```diff
@@ -69,12 +93,12 @@ void RPCHitCleaner::run(const edm::EventSetup& c) {
   for( auto chamber = m_inrpcDigis.begin(); chamber != m_inrpcDigis.end(); ++chamber ){
         RPCDetId detid = (*chamber).first;
         if(detid.region()!=0 ) continue; //Region = 0 Barrel
-        int strips[5] = {0};
+        BxToStrips strips;
```
[the remark this answers is not in the document]
Comment (Dr15Jones) [reply L1Trigger/L1TTwinMux/src/RPCHitCleaner.cc]: The first bug I found was this one
https://github.com/cms-sw/cmssw/pull/21151/files#diff-bb4facdb289f9efc85f12aad556325e7R124

It was a missing ‘+3’ in the array index calculation.

[the remark this answers is not in the document]
Comment (Dr15Jones) [reply L1Trigger/L1TTwinMux/src/RPCHitCleaner.cc]: Ignore that, I shouldn’t reply to messages before having coffee in the morning.

**Comment by mrodozov** [comment]
I ran the failing wf's on arm with this changes and I confirm this PR is fixing most of the fails.

File: L1Trigger/L1TTwinMux/src/RPCHitCleaner.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidlange6) [reply L1Trigger/L1TTwinMux/src/RPCHitCleaner.cc]: seems @rekovic won't comment so we'll go forward here. (and push in 94x)

**Comment by davidlange6** [comment]
merge

**Pull request merged**